plugins/discovery-ec2/src/main/java/org/elasticsearch/cloud/aws/AwsEc2ServiceImpl.java

```diff
@@ -107,14 +85,14 @@ public synchronized AmazonEC2 client() {
         }
 
         // #155: we might have 3rd party users using older EC2 API version
-        String awsSigner = settings.get(CLOUD_EC2.SIGNER, settings.get(CLOUD_AWS.SIGNER));
-        if (awsSigner != null) {
+        String awsSigner = CLOUD_EC2.SIGNER_SETTING.get(settings);
+        if (Strings.hasText(awsSigner)) {
             logger.debug("using AWS API signer [{}]", awsSigner);
             try {
                 AwsSigner.configureSigner(awsSigner, clientConfiguration);
             } catch (IllegalArgumentException e) {
                 logger.warn("wrong signer set for [{}] or [{}]: [{}]",
```

Comment:
I am not sure whether we want to be lenient when a wrong signer is set. What happens in that case?